### PR TITLE
Bulk-fastt-enable-latest-earliest

### PR DESCRIPTION
**Pull request by danyx23**
This PR is part of #1224 - it adds a new operator so that on time fields like maxTime you can query for entries "latest" or "earliest"

### CODE REVIEW

**Review by danielgavrilov — APPROVED** [review]
Looks good!

I am doing just a basic sense-check review since I don't have a great handle of the codebase here, let me know if it needs more than that.

File: adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx [hunk 1]
```diff
@@ -339,6 +340,38 @@ export function SExpressionToJsonLogic(
...
+        const isEarliest =
+            filterTree.properties.value.length === 1 &&
+            filterTree.properties.value[0] === "earliest"
+        const isEqual =
+            filterTree.properties.operator === "equal" ||
+            filterTree.properties.operator === "select_equals"
+        if (isLatest && isEqual) {
+            filterTree.properties.operator = "is_latest"
+        } else if (isEarliest && isEqual) {
+            filterTree.properties.operator = "is_earliest"
```
Comment (ikesau) [inline adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx:371]: I'd deconstruct to DRY this up a bit.

```
const { properties: { value }, operator }  = filterTree

if (value.length !== 1) return

const isLatest = value[0] === "latest"

const isEarliest = value[0] === "earliest"

const isEqual = operator === "equal" || operator === "select_equals"
```
Up to you! Still getting a sense for the preferred idioms on this project.

**Review by ikesau — APPROVED** [review]
Same as gavrilov! Assuming the logic's good, just a few style suggestions.

File: adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx [hunk 2]
```diff
@@ -488,12 +534,17 @@ export function fieldDescriptionToFilterPanelFieldConfig(
...
         .with(EditorOption.mappingEditor, () => undefined)
         .with(EditorOption.numeric, () => "number")
+        .with(EditorOption.numericWithLatestEarliest, () => "number")
         .with(EditorOption.primitiveListEditor, () => undefined)
         .with(EditorOption.textarea, () => "text")
         .with(EditorOption.textfield, () => "text")
         .exhaustive()
 
-    if (widget !== undefined)
+    if (widget !== undefined) {
```
[the remark this answers is not in the document]
Comment (danyx23) [reply adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx:543]: I think the cognitive overhead of falsy checks is not great - you have to consider the type to know the behaviour. This is why I prefer to check explicitly for undefined (or lodash.isNil when either null or undefined).

File: adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx (hunk 1 above)
Comment (danyx23) [reply adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx:371]: Yeah true, that reads nicer. My brain always goes into legacy mode with these things in TS as in an ML language could write it like this:
```fsharp
match (filterTree.properties) with
| {value = ["latest"], operator = "equal"}
| {value = ["latest"], operator = "select_equal"} -> "is_latest"
| {value = ["earliest"], operator = "equal"}
| {value = ["earliest"], operator = "select_equal"} -> "is_earliest"
```

**Comment by danyx23** [comment]
Thanks @ikesau and @danielgavrilov for your reviews!

**Pull request merged**